Lib/xml/etree/ElementPath.py

```diff
@@ -188,14 +190,21 @@ def select(context, result):
                 if elem.find(tag) is not None:
                     yield elem
         return select
-    if signature == "-='" and not re.match(r"\-?\d+$", predicate[0]):
-        # [tag='value']
+    if signature == ".='" or (signature == "-='" and not re.match(r"\-?\d+$", predicate[0])):
+        # [.='value'] or [tag='value']
         tag = predicate[0]
         value = predicate[-1]
-        def select(context, result):
-            for elem in result:
-                for e in elem.findall(tag):
-                    if "".join(e.itertext()) == value:
+        if tag:
+            def select(context, result):
+                for elem in result:
+                    for e in elem.findall(tag):
+                        if "".join(e.itertext()) == value:
+                            yield elem
+                            break
+        else:
+            def select(context, result):
+                for elem in result:
+                    if "".join(elem.itertext()) == value:
                         yield elem
                         break
```

Comment:
I think there shouldn't be a `break`. Add a test for this.

Reply:
This is inside of a predicate, so we only care about "found" or "not found". The break was there before.

Reply:
The break was there before for breaking an internal loop. `findall()` should find all elements, not the first one.

Reply:
To clarify: the break is correct.

Reply:
Ah, you're right. :)

Reply:
Fixed.